### ISSUE TITLE
Don't blockingly load songs at startup

### ISSUE THREAD

**Issue by xeruf** [puddletag/puddletag#915]
### What's missing?

When I start puddletag, I have to wait a minute or so because I have a lot of songs.

### Your solution

Own database cache, or even interoperating with a tool like mpd or beets for indexing

### Any alternatives?

Loading in the background, with at least the main UI already usable (so I can for example switch the folder)

**Comment by xeruf** [comment]
same for a search query - the whole UI freezes for a few seconds when I run a query, that is not something I am used to - tools like everything or mixxx have a database that produces live results as you type with thousands of entries

**Comment by rambit** [comment]
I was about to make a new issue exactly about this.

When I load a folder of mp3 files from an NFS drive, the popup "Loading X of Y" loads perhaps 2 songs per second, so that's about a minute for over 100 songs. `scp` from the same NFS drive gives me about 10 mb/s which tells me that Puddletag probably loads the entire file to extract the metadata. Is that the only way?

If I cancel and restart Puddletag, at least it seems to have cached the previously loaded songs, but only till after I restart the computer.

My primary use for Puddletag is actually being a loader for my player, as I can sort on tags and select what I want to listen to, but the slow loading times makes it a bit tedious.

**Comment by rambit** [comment]
You are right. It was NFS which I use at v4. Setting a mount option `rsize` of 1MB to load data in smaller chunks improved the loading speed.

**Comment by thefoster** [comment]
> My primary use for Puddletag is actually being a loader for my player, as I can sort on tags and select what I want to listen to, but the slow loading times makes it a bit tedious.

I suggest using the tools for what they are intended (the unix way).
E.g. MP3Diags for fixing errors, puddletag for setting/tidying up the tags and maybe kid3-cli for bulk fixing on the shell what puddletag can't (i once had an issue with embedded cover images with type "other" rather that "cover front", so some players didn't recognize it).

Loading all my audiobooks and music over NFSv4 at once takes about 4h at 25MB/s, so i don't do that very often, just with new files and when i find e.g. mistyped artist or album name.

Once your MP3s are tagged well (e.g. genre), almost every player can handle them very smoothly, for example elisa, amarok/clementine and even the streaming players like volumio.

**Comment by sandrotosi** [comment]
> You are right. It was NFS which I use at v4. Setting a mount option `rsize` of 1MB to load data in smaller chunks improved the loading speed.

based on this, given it appears to be resolved, im closing this issue

**Comment by xeruf** [comment]
the original issue is not resolved - loading a bigger music database even on a fast disk takes a while
